src/jdk.jlink/share/classes/jdk/tools/jlink/internal/runtimelink/RuntimeImageLinkException.java

```diff
@@ -35,7 +35,6 @@ public class RuntimeImageLinkException extends RuntimeException {
     private static final long serialVersionUID = -1848914673073119403L;
 
     public static enum Reason {
-        PATCH_MODULE,  /* link exception due to patched module */
```

Comment:
Given there is only one reason to throw `RuntimeImageLinkException`, I wonder if `RuntimeImageLinkException` is still needed.  Is RILE needed because checking the mismatch is done in stream?   Maybe `JRTArchive::addNonClassResources` can return true/false to indicate if the file is modified.

Reply:
The updated patch has RuntimeImageLinkException removed. UncheckedIOException is being used instead.